cardano-cli/src/Cardano/CLI/Compatible/Exception.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Cardano.CLI.Compatible.Exception
+  ( CustomCliException (..)
+  , throwCliError
```

Comment:
I don't think this should be part of the API of this module. This function isn't useful on its own.
```suggestion
```

Reply:
What do you mean "attach a call stack manually"? I generated the exception [here](https://github.com/IntersectMBO/cardano-cli/pull/1033#discussion_r1954497705) with `throwCliError`.

Reply:
It is now fine with implicit call stack in `CustomCliException`. Previously when you had to manually provide `callStack` to the constructor it would just show up the call stack to the nearest `HasCallStack` constraint. I mean if you would do this previously in some function:
```haskell
foo :: HasCallStack => RIO e ()
foo = do
  ...
  ...
  throwCliError $ CustomCliException $ Dummy
```
you'd get the location of a `foo` call, not where the `Dummy` was thrown.